lib/presentation/common_widgets/base/base_page_stateless.dart

```diff
@@ -0,0 +1,10 @@
+import 'package:flutter/material.dart';
+
+import 'base_page_mixin.dart';
+
+abstract class BasePageStateless extends StatelessWidget with BasePageMixin {
+  const BasePageStateless({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) => buildPage(context);
+}
```

Comment:
i agree with your above explanations my point was how we use different parent classes of river pod
ConsumerWidget, ConsumerStatefulWidget, ConsumerStatefulElement etc.
because this is river pod oss
may be need more bases classes for each type?

any way you can ignore this for now